core/src/main/java/org/apache/iceberg/FindFiles.java

```diff
@@ -103,7 +103,7 @@ public Builder asOfTime(long timestampMillis) {
       // case, there is no valid snapshot to read.
       Preconditions.checkArgument(lastSnapshotId != null,
           "Cannot find a snapshot older than %s",
-          DATE_FORMAT.format(LocalDateTime.ofInstant(Instant.ofEpochMilli(timestampMillis), ZoneId.systemDefault())));
```

Comment:
Another slight change in behavior here ... I would probably save these for another PR unless we really need it here

Reply:
yes, this was an error reported from the newer error-prone checks

> com.palantir.baseline-error-prone uses a newer error-prone version (and does additional checks), so a few changes were necessary to fix JavaTimeSystemDefaultTimeZone / ConsistentOverrides errors

Reply:
could we just skip that rule? We do for some other error prone rules. I'm fine with maybe making this change later just seems unrelated